server/routes/record-attendance/handlers/attendanceList.ts

```diff
@@ -39,7 +39,7 @@ export default class AttendanceListRoutes {
             .getScheduledEventsForPrisoners(toDate(instance.date), prisonerNumbers, user)
             .then(response => [
               ...response.activities,
-              ...response.appointments,
+              ...response.appointments.filter(app => !app.deleted),
```

Comment:
There's a comment on the related API change which will mean the API doesn't return deleted appointments. If that's implemented, this filter and the other instances of it can be removed to simplify the logic paths and test scenarios

Reply:
API now excluding deleted appointment instances